examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -440,16 +503,63 @@ def _cumulated_claims(y_true, y_pred, exposure):
 ax.legend(loc="upper left")
 
 ##############################################################################
-# This plot reveals that the random forest model is slightly better at ranking
-# policyholders by risk profiles even if the absolute value of the predicted
-# expected frequencies are less well calibrated than for the linear Poisson
-# model.
+# As expected, the dummy regressor is unable to correctly rank the samples and
+# therefore performs the worst on this plot.
+#
+# The tree-based model is significantly better at ranking policyholders by risk
+# while the two linear models perform similarly.
 #
 # All three models are significantly better than chance but also very far from
 # making perfect predictions.
 #
 # This last point is expected due to the nature of the problem: the occurrence
 # of accidents is mostly dominated by circumstantial causes that are not
-# captured in the columns of the dataset or that are indeed random.
+# captured in the columns of the dataset and can indeed be considered as purely
+# random.
+#
+# The linear models assume no interactions between the input variables which
+# likely causes under-fitting. Inserting a polynomial feature extractor
+# (func:`sklearn.preprocessing.PolynomialFeatures`) indeed increases their
+# discrimative power by 2 points of Gini index. In particular it improves the
+# ability of the models to identify the top 5% riskiest profiles.
+#
+# Main takeaways
+# --------------
+#
+# - The performance of the models can be evaluted by their ability to yield
+#   well-calibrated predictions and a good ranking.
+#
+# - The Gini index reflects the ability of a model to rank predictions
+#   irrespective of their absolute values, and therefore only assess their
+#   ranking power.
+#
+# - The calibration of the model can be assessed by plotting the mean observed
+#   value vs the mean predicted value on groups of test samples binned by
+#   predicted risk.
+#
+# - The least squares loss (along with the implicit use of the identity link
+#   function) of the Ridge regression model seems to cause this model to be
+#   badly calibrated. In particular, it tends to underestimate the risk and can
+#   even predict invalid negative frequencies.
+#
+# - Using the Poisson loss with a log-link can correct these problems and lead
+#   to a well-calibrated linear model.
+#
+# - Despite the improvement in calibration, the ranking power of both linear
+#   models are comparable and well below the ranking power of the Gradient
+#   Boosting Regression Trees.
+#
+# - The non-linear Gradient Boosting Regression Trees model does not seem to
```

Comment:
We didn't evaluate this one right? I guess it's fine leaving the entry but we should precise we left it out

Reply:
Good point I need to update this bullet point since I switched to the Poisson loss for HGBRT.